ai_ref_kits/multimodal_ai_visual_generator/main.py

```diff
@@ -1,17 +1,209 @@
+    with open(config_file, "r") as f:
+        return yaml.safe_load(f)
+
+# --- Helper: Parse LLM output into scenes ---
+def parse_scenes(output_text: str, config: dict) -> list[str]:
+    scenes = []
+    current_scene = ""
+    for line in output_text.split("\n"):
+        clean = line.strip()
+        if clean.lower().startswith(config["scene_prefix"].lower()):
+            if current_scene:
+                scenes.append(current_scene.strip())
+            parts = clean.split(":", 1)
+            if len(parts) == 2:
+                current_scene = parts[1].strip()
+        elif clean:
+            current_scene += " " + clean
+    if current_scene:
+        scenes.append(current_scene.strip())
+    return scenes
+
+# --- Helper: Clean scenes and apply suffix ---
+def finalize_scenes(scenes: list[str], config: dict) -> list[str]:
+    suffixes = config["scene_suffixes"]
+    max_words = config["max_words_per_scene"]
+    final_scenes = []
+    for i, scene in enumerate(scenes[:4]):
+        words = scene.split()
+        trimmed = " ".join(words[:max_words])
+        suffix = suffixes[i] if i < len(suffixes) else suffixes[-1]
+        if suffix.lower() not in trimmed.lower():
+            if not trimmed.endswith((".", "!", "?")):
+                trimmed += "."
+            trimmed += " " + suffix
+        final_scenes.append(trimmed.strip())
+    while len(final_scenes) < 4:
+        fallback = config["fallback_scene"]
+        fallback_suffix = suffixes[len(final_scenes)] if len(suffixes) > len(final_scenes) else suffixes[-1]
+        final_scenes.append(fallback + ". " + fallback_suffix)
+    return final_scenes
+
+# ---------- LLM Endpoint (Story Splitter) ---------
+@app.post("/generate_story_prompts")
+def generate_story_prompts(request: StoryRequest, req: Request):
+    if not llm_pipe:
+        return JSONResponse(status_code=503, content={"error": "LLM model not available. Please export it before using this endpoint."})
+
+    config = load_story_config(req)
+    instruction = config["instruction_template"].replace("{user_prompt}", request.prompt)
+
+    output = []
+    def streamer(subword):
+        sys.stdout.write(subword)
+        sys.stdout.flush()
+        output.append(subword)
+        return False
+
+    _ = llm_pipe.generate(instruction, llm_config, streamer)
+    full_output = "".join(output)
+
+    parsed_scenes = parse_scenes(full_output, config)
+    final_scenes = finalize_scenes(parsed_scenes, config)
+
+    return {"scenes": final_scenes}
+
+# ---------- Image Model Endpoint (Image Generator) ----------
+@app.post("/generate_images")
+def generate_image(request: PromptRequest):
+    if not image_pipe:
+        return JSONResponse(status_code=503, content={"error": "Image model not available. Please export it before using this endpoint."})
+
+    prompt = request.prompt
+    height = 512
+    width = 512
+    seed = random.randint(1_000_000_000, 4_294_967_295)

```

Comment:
Why not 0 to np.iinfo(np.int32).max?

Reply:
I tested for single-digit seed value for multiple scenarios and generated well especially for the Flux model. So I incorporated the suggested feedback. We can revise this in case issue arises with model generation in future.